src/admin/index.js

```diff
@@ -426,6 +426,82 @@ module.exports = ({
       .pop()
   }
 
+  /**
+   * @param {string} groupId
+   * @param {string[]]} topics
+   * @param {boolean} [resolveOffsets=false]
+   * @return {Promise}
+   */
+  const fetchConsumerGroupOffsets = async ({ groupId, topics, resolveOffsets = false }) => {
+    if (!groupId) {
+      throw new KafkaJSNonRetriableError(`Invalid groupId ${groupId}`)
+    }
+
+    const coordinator = await cluster.findGroupCoordinator({ groupId })
+
+    let consumerOffsets
+    if (topics.length) {
+      const topicsToFetch = Promise.all(
+        topics.map(async topic => {
+          const partitions = await findTopicPartitions(cluster, topic)
+          const partitionsToFetch = partitions.map(partition => ({ partition }))
+          return { topic, partitions: partitionsToFetch }
+        })
+      )
+      const { responses } = await coordinator.offsetFetch({
+        groupId,
+        topicsToFetch,
+      })
+      consumerOffsets = responses
+    } else {
+      const { responses } = await coordinator.offsetFetch({ groupId })
+      consumerOffsets = responses
+    }
+
+    if (resolveOffsets) {
+      consumerOffsets = Promise.all(
+        consumerOffsets.map(async ({ topic, partitions }) => {
+          const indexedOffsets = indexByPartition(await fetchTopicOffsets(topic))
+          const recalculatedPartitions = partitions.map(({ offset, partition, ...props }) => {
+            let resolvedOffset = offset
+            if (Number(offset) === EARLIEST_OFFSET) {
+              resolvedOffset = indexedOffsets[partition].low
+            }
+            if (Number(offset) === LATEST_OFFSET) {
+              resolvedOffset = indexedOffsets[partition].high
+            }
+            return {
+              partition,
+              offset: resolvedOffset,
+              ...props,
+            }
+          })
+
+          return {
+            topic,
+            partitions: recalculatedPartitions,
+          }
+        })
+      )
+
+      await Promise.all(
+        consumerOffsets.map(({ topic, partitions }) => setOffsets({ groupId, topic, partitions }))
+      )
+    }
+
+    return consumerOffsets
+      .filter(response => !topics.length || response.topic in topics)
```

Comment:
Isn't `topics` an array? `in` checks for the existence of a key in an object, not whether an item is in an array. Maybe you meant `topics.includes(response.topic)`?

Reply:
Yeah it was just a draft PR and I'm not super familiar with javascript :)
I've removed this line altogether since I realized why would response not always include exactly the topics we requested?